### PR TITLE
ADD throw on runtime errors

### PR DESCRIPTION
**Pull request by ndelangen**
fixes: https://github.com/chromaui/chromatic-cli/issues/75#issuecomment-572265026

### CODE REVIEW

File: bin/tester/runtimes.js [hunk 1]
```diff
@@ -89,11 +89,20 @@ export default async function getRuntimeSpecs(url, { verbose = false } = {}) {
       );
     }
 
-    console.log(dedent`
-        This may lead to some stories not working right or getting detected by Chromatic
-        We suggest you fix the errors, but we will continue anyway..
-        ${separator}
-      `);
+    if (errors.length && !log.level.match(/silent/)) {
```
Comment (tmeasday) [inline bin/tester/runtimes.js]: Are we going to allow them to continue if log level is silent? That seems incorrect, maybe?

File: bin/tester/runtimes.js [hunk 2]
```diff
@@ -89,11 +89,20 @@ export default async function getRuntimeSpecs(url, { verbose = false } = {}) {
...
+      console.log(dedent`
+          This very likely caused some stories not working right or getting detected by Chromatic
+          Please fix the errors, we can't continue..
+          ${separator}
+        `);
+      throw new Error('Errors detected in storybook runtime');
+    } else if (warnings.length && log.level.match(/verbose/)) {
```
Comment (tmeasday) [inline bin/tester/runtimes.js]:
```suggestion
    } else if (hasVisibileWarnings) {
```

Can we pull this out to a variable? I guess we are using it 3 times now.

File: bin/tester/runtimes.js (hunk 1 above)
Comment (ndelangen) [reply bin/tester/runtimes.js]: 👍

File: bin/tester/runtimes.js (hunk 2 above)
Comment (ndelangen) [reply bin/tester/runtimes.js]: sure

File: bin/tester/runtimes.js [hunk 3]
```diff
@@ -47,17 +47,16 @@ export default async function getRuntimeSpecs(url, { verbose = false } = {}) {
...
   });
 
-  // If the app logged something to console.error, it's probably, but not definitely an issue.
-  // See https://github.com/hichroma/chromatic/issues/757
-  if (
-    (errors.length && !log.level.match(/silent/)) ||
-    (warnings.length && log.level.match(/verbose/))
-  ) {
+  const hasErrors = errors.length;
+  const hasVisibileErrors = hasErrors && log.level.match(/verbose/);
```
Comment (tmeasday) [inline bin/tester/runtimes.js]:
```suggestion
  const hasVisibleErrors = hasErrors && log.level.match(/verbose/);
```

File: bin/tester/runtimes.js [hunk 4]
```diff
@@ -89,11 +88,20 @@ export default async function getRuntimeSpecs(url, { verbose = false } = {}) {
       );
     }
 
-    console.log(dedent`
-        This may lead to some stories not working right or getting detected by Chromatic
-        We suggest you fix the errors, but we will continue anyway..
-        ${separator}
-      `);
+    if (hasErrors) {
```
Comment (tmeasday) [inline bin/tester/runtimes.js]: If they have suppressed errors we probably need to say something here.

I wonder if we should just drop the silent behaviour? Is it useful to not show errors if we are going to stop?

Comment (tmeasday) [reply bin/tester/runtimes.js]: OK, I took a look and silent is only used via the `DISABLE_LOGGING` env var, which we use in tests, so we don't need to worry to much about it.

Comment (ndelangen) [reply bin/tester/runtimes.js]: cool

**Pull request merged**